frontend/middleware_validatesubscription.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/api/subscription"
+)
+
+type SubscriptionStateMuxValidator struct {
+	cache *Cache
+}
+
+func NewSubscriptionStateMuxValidator(c *Cache) *SubscriptionStateMuxValidator {
+	return &SubscriptionStateMuxValidator{
+		cache: c,
+	}
+}
+
+func (s *SubscriptionStateMuxValidator) MiddlewareValidateSubscriptionState(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
```

Comment:
Maybe we can name this method just _ValidateSubscriptionState_ to avoid redundancy.

Reply:
I prefer to keep `Middleware` in the name, so it aligns with all other `Middleware`

Reply:
Makes sense